staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/coerce.go

```diff
@@ -18,40 +18,78 @@ package objectmeta
 
 import (
 	"fmt"
+	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	utiljson "k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/klog/v2"
 )
 
 // GetObjectMeta does conversion of JSON to ObjectMeta. It first tries json.Unmarshal into a metav1.ObjectMeta
 // type. If that does not work and dropMalformedFields is true, it does field-by-field best-effort conversion
 // throwing away fields which lead to errors.
 func GetObjectMeta(obj map[string]interface{}, dropMalformedFields bool) (*metav1.ObjectMeta, bool, error) {
+	meta, found, _, err := GetObjectMetaStrict(obj, dropMalformedFields, false)
+	return meta, found, err
+}
+
+// GetObjectMetaStrict allows for strict unmarshaling of the metadata bytes, returning any unknown field paths
+// it encounters
+func GetObjectMetaStrict(obj map[string]interface{}, dropMalformedFields, strict bool) (*metav1.ObjectMeta, bool, []string, error) {
 	metadata, found := obj["metadata"]
 	if !found {
-		return nil, false, nil
+		return nil, false, nil, nil
 	}
 
 	// round-trip through JSON first, hoping that unmarshalling just works
 	objectMeta := &metav1.ObjectMeta{}
 	metadataBytes, err := utiljson.Marshal(metadata)
 	if err != nil {
-		return nil, false, err
+		return nil, false, nil, err
 	}
-	if err = utiljson.Unmarshal(metadataBytes, objectMeta); err == nil {
-		// if successful, return
-		return objectMeta, true, nil
+	if strict {
```

Comment:
Does this correctly handle the `Warn` case where we want to *know* the unknown paths, but don't want to *fail* on any strict errors when dropMalformedFields is true? it looks like this loses the field-by-field decoding tolerating malformed fields when dropMalformedFields is true

Reply:
looks like this comment thread is hidden because the diff is outdated, hoisted this to https://github.com/kubernetes/kubernetes/pull/109494/files#r921576674